### PR TITLE
CK group forward convolution integration 

### PR DESCRIPTION
**Pull request by iq136boy**
Integrate the latest ck group forward convolution into miopen

### CODE REVIEW

File: src/solver.cpp [hunk 1]
```diff
@@ -535,6 +535,9 @@ inline SolverRegistrar::SolverRegistrar(IdRegistryData& registry)
         registry, ++id, ConvHipImplicitGemmFwdXdlops{}, miopenConvolutionAlgoImplicitGEMM);
     RegisterWithSolver(
         registry, ++id, ConvHipImplicitGemmBwdXdlops{}, miopenConvolutionAlgoImplicitGEMM);
+    RegisterWithSolver(
```
Comment (DrizztDoUrden) [inline src/solver.cpp]: This should go to the end of the function.

File: test/gtest/group_conv_fwd.cpp [hunk 2]
```diff
@@ -0,0 +1,99 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2022 Advanced Micro Devices, Inc.
```
Comment (DrizztDoUrden) [inline test/gtest/group_conv_fwd.cpp]:
```suggestion
 * Copyright (c) 2023 Advanced Micro Devices, Inc.
```

Comment (DrizztDoUrden) [inline test/gtest/group_conv_fwd.cpp]:
```suggestion
 * Copyright (c) 2023 Advanced Micro Devices, Inc.
```

File: src/solver/conv_hip_implicit_gemm_grouped_fwd_xdlops.cpp [hunk 3]
```diff
@@ -0,0 +1,467 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2022 Advanced Micro Devices, Inc.
```
Comment (DrizztDoUrden) [inline src/solver/conv_hip_implicit_gemm_grouped_fwd_xdlops.cpp]:
```suggestion
 * Copyright (c) 2023 Advanced Micro Devices, Inc.
```

**Review by DrizztDoUrden — CHANGES_REQUESTED** [review]

File: src/solver.cpp (hunk 1 above)
Comment (iq136boy) [reply src/solver.cpp]: Fixed

**Review by DrizztDoUrden — DISMISSED** [review]

File: test/gtest/group_solver.hpp [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
```
Comment (averinevg) [inline test/gtest/group_solver.hpp:25]: License has been included twice

File: test/gtest/group_solver.hpp [hunk 5]
```diff
@@ -0,0 +1,214 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2022 Advanced Micro Devices, Inc.
```
Comment (averinevg) [inline test/gtest/group_solver.hpp]:
```suggestion
 * Copyright (c) 2023 Advanced Micro Devices, Inc.
```

**Review by averinevg — CHANGES_REQUESTED** [review]

**Review by averinevg — DISMISSED** [review]

**Review by JehandadKhan — DISMISSED** [review]

**Comment by JehandadKhan** [comment]
@DrizztDoUrden and @averinevg  Can we give this PR a second look before we merge it?

**Review by averinevg — DISMISSED** [review]

**Comment by junliume** [comment]
@iq136boy could you help to resolve the conflict?

**Review by averinevg — DISMISSED** [review]

**Review by JehandadKhan — DISMISSED** [review]

**Comment by junliume** [comment]
I restarted CI process for this PR

**Comment by iq136boy** [comment]
> I restarted CI process for this PR

@junliume there is some inconsistence between this PR and Evgnenii's PR merged yesterday. I am working now to fixed it.

**Comment by iq136boy** [comment]
@JehandadKhan and @junliume CI passed.

**Review by junliume — APPROVED** [review]

File: src/solver/conv_hip_implicit_gemm_grouped_fwd_xdlops.cpp [hunk 6]
```diff
@@ -0,0 +1,467 @@
...
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = ctx;
+    std::ignore = problem;
+    return false;
+#else
+    if(miopen::IsDisabled(MIOPEN_DEBUG_GROUP_CONV_IMPLICIT_GEMM_HIP_FWD_XDLOPS{}))
+        return false;
+    if(miopen::IsEnabled(MIOPEN_DEBUG_CONVOLUTION_DETERMINISTIC{}))
+        return false;
```
Comment (atamazov) [inline src/solver/conv_hip_implicit_gemm_grouped_fwd_xdlops.cpp:401]: [Correctness] This is wrong as it does not obey the attribute set in the convolution descriptor.
```C++
    if(problem.conv_problem.GetConv().attribute.deterministic)
        return false;
```

**Review by atamazov — COMMENTED** [review]
the "deterministic" attribute is not observed.

File: src/solver/conv_hip_implicit_gemm_grouped_fwd_xdlops.cpp [hunk 7]
```diff
@@ -0,0 +1,467 @@
...
+                RunCKSolution<float>(handle, primitive_parameters, problem, config);
+                break;
+            case miopenInt8:
+                RunCKSolution<int8_t>(handle, primitive_parameters, problem, config);
+                break;
+            case miopenInt32:
+            case miopenInt8x4:
+            case miopenBFloat16:
+            case miopenDouble: break;
+            }
```
Comment (atamazov) [inline src/solver/conv_hip_implicit_gemm_grouped_fwd_xdlops.cpp:459]: [Performance] The same as https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2286#discussion_r1286007620. Computations that depend on problem should be moved from Invoker to InvokerFactory.

Comment (atamazov) [reply src/solver/conv_hip_implicit_gemm_grouped_fwd_xdlops.cpp:459]: @iq136boy 

> [Performance] The same as [#2286 (comment)](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2286#discussion_r1286007620). Computations that depend on problem should be moved from Invoker to InvokerFactory.

Original comment updated: added notice that computations can be moved even higher, into `GetSolution`.

Comment (atamazov) [reply src/solver/conv_hip_implicit_gemm_grouped_fwd_xdlops.cpp:459]: [resolved]

File: src/solver/conv_hip_implicit_gemm_grouped_fwd_xdlops.cpp (hunk 6 above)
Comment (atamazov) [reply src/solver/conv_hip_implicit_gemm_grouped_fwd_xdlops.cpp:401]: [resolved]

File: src/ocl/convolutionocl.cpp [hunk 8]
```diff
@@ -100,8 +100,7 @@ static inline void ValidateGroupCount(const TensorDescriptor& xDesc,
              (wDesc.GetLayout_t() == miopenTensorNCHWc8)) &&
             (xDesc.GetLengths()[1] / conv.group_count != wDesc.GetLengths()[1])) ||
            ((wDesc.GetLayout_t() == miopenTensorCHWNc4 ||
-             wDesc.GetLayout_t() == miopenTensorCHWNc8) &&
-            (xDesc.GetLengths()[1] / conv.group_count != wDesc.GetLengths()[0])))
+             wDesc.GetLayout_t() == miopenTensorCHWNc8)))
```
Comment (atamazov) [inline src/ocl/convolutionocl.cpp:103]: @iq136boy @JehandadKhan Why this change is necessary? https://github.com/ROCmSoftwarePlatform/MIOpen/labels/urgency_high
